kusd/validator/election.go

```diff
@@ -76,19 +79,29 @@ func NewVotingSystem(eventMux *event.TypeMux, signer types.Signer, electionNumbe
 }
 
 func (vs *VotingSystem) NewRound() {
-	vs.votesPerRound[vs.round] = NewVotingTables(vs.eventMux, vs.signer, vs.electionNumber, vs.round, vs.voters)
+	vs.votesPerRound[vs.round] = NewVotingTables(vs.eventMux, vs.voters)
 }
 
 // Add registers a vote
-func (vs *VotingSystem) Add(vote *types.Vote, local bool) (bool, error) {
-	// @TODO (rgeraldes) - validation
+func (vs *VotingSystem) Add(vote *types.Vote) error {
 	votingTable := vs.getVoteSet(vote.Round(), vote.Type())
-	votingTable.Add(vote, local)
 
-	return false, nil
+	signedVote, err := types.NewSignedVote(vs.signer, vote)
```

Comment:
Instead of NewSignedVote I think that vote.AsSignedVote makes more sense - but if the purpose is to get the sender I think that we should look for a different solution. It seems that we are creating a new vote.

Reply:
yeah, I commented myself on this one :)

it will prob go away as I refactor outer layers.

Yes, `Vote` should prob have `Address`? 

if I have an instance of `Vote` then it's a valid one (signed) so the consumer of vote doesn't need to worry about checking is valid.

Reply:
Can we add Address to Vote and on decoding a vote fail if signature is wrong

Reply:
transaction does something like that from field - https://github.com/ethereum/go-ethereum/blob/master/core/types/transaction.go#L54

Reply:
will something similiar on next refactor